### PR TITLE
Raised timeout to 15 seconds

### PR DESCRIPTION
**Pull request by rishimo**
### 🤔 Why?

[SC24473](https://app.shortcut.com/metaphor-data/story/24473/implement-notion-api-rate-limiting-higher-timeout)

After a couple title retrieval fails, the crawler fails to make any subsequent connections to the Notion API. Increasing the timeout on title seems to allow for enough time for a successful re-connection after failures. 

### 🤓 What?

Changes the _get_title timeout from 5 to 15 seconds.

### 🧪 Tested?

Tested locally and works with the test Notion instance.

### ☑️ Checks

- [x] My PR contains actual code changes, and I have updated the version number in `pyproject.toml`.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -120,6 +124,7 @@ power_bi = ["msal", "msgraph-beta-sdk", "sql-metadata"]
 redshift = ["asyncpg", "sqllineage"]
 s3 = ["fastavro", "more-itertools", "parse"]
 snowflake = ["snowflake-connector-python", "sql-metadata"]
+static_web = ["beautifulsoup4", "llama-hub", "lxml"]
```
Comment (prith189) [inline pyproject.toml:127]: These would be part of different PR?

Comment (rishimo) [reply pyproject.toml:127]: Oops yes they are for https://github.com/MetaphorData/connectors/pull/771 😬

**Review by mars-lan — APPROVED** [review]
